### PR TITLE
Add a note about CrateDB not supporting `UNIQUE` column constraints

### PR DESCRIPTION
**Pull request by norosa**
Closes <https://github.com/crate/crate/issues/11248>

### CODE REVIEW

File: docs/sql/general/constraints.rst [hunk 1]
```diff
@@ -165,6 +165,11 @@ The supported column constraints are:
 
 - :ref:`check_constraint`
 
+.. NOTE::
+
+    CrateDB does not support the ``UNIQUE`` column constraint because it is
+    impossible to enforce with an :ref:`eventually-consistent
+    <concept-consistency>` consistency model.
```
Comment (seut) [inline docs/sql/general/constraints.rst:172]: It's not impossible but not easy indeed. I suggest to remove any reasoning.

**Comment by mfussenegger** [comment]
I'm not sure if we should start documenting SQL standard features that we lack like this. Where do you draw the line? Why is the UNIQUE constraint mentioned but the other `CREATE TABLE` options that we miss not?

The SQL features table already gives an overview of what's there or not, isn't that sufficient?

**Comment by norosa** [comment]
fair enough! ty both for the feedback!

**Comment by mfussenegger** [comment]
To follow up on my earlier comment:

I think a compromise might be a general "SQL Standard differences" section that talks about the eventual consistency/ACID/referential integrity aspect on a higher level and then maybe mentions some of the implications.

**Pull request closed without merging**